src/kern_db.cpp

```diff
@@ -53,23 +61,13 @@ KernDb::KernDb(const std::string& filename_,
             MIOPEN_THROW(miopenStatusInternalError);
         MIOPEN_LOG_I2("Database created successfully");
     }
-    if(!dbInvalid)
+    if(!CheckTableColumns(KernelConfig::table_name(), KernelConfig::FieldNames()))
     {
-        if(!CheckTableColumns(KernelConfig::table_name(), KernelConfig::FieldNames()))
-        {
-            std::ostringstream ss;
-            ss << "Invalid fields in table: " << KernelConfig::table_name()
-               << " disabling access to " << filename;
-            MIOPEN_LOG_W(ss.str());
-            dbInvalid = true;
-        }
-    }
-    else
-    {
-        if(filename.empty())
-            MIOPEN_LOG_I("database not present");
-        else
-            MIOPEN_LOG_I(filename + " database invalid");
+        std::ostringstream ss;
+        ss << "Invalid fields in table: " << KernelConfig::table_name() << " disabling access to "
+           << filename;
+        MIOPEN_LOG_W(ss.str());
+        dbInvalid = true;
```

Comment:
Instead of using a separate `dbInvalid` flag, why cant you just set the `ptrDb` to null when its invalid? It seems like it doesnt make sense to keep the connection open if the db is invalid or disabled.

Reply:
@JehandadKhan please follow up with another PR.